### PR TITLE
Tests: Add response error along with unexpected status in report

### PR DESCRIPTION
**Pull request by fflorent**
## Description

Show the error message the server returned in the response when the status is not the one expected.

Note: I proposed to help the user so they can see more details in log. The last line displays a URL. I am not very sure of the relevancy, feedback welcome.

## Has this been tested?

- [x] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because this PR does not update server code
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: shared/models/http/http-error-codes.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@
  *
  * WebDAV and other codes useless with regards to PeerTube are not listed.
  */
-export const enum HttpStatusCode {
+export enum HttpStatusCode {
```
Comment (Chocobozzz) [inline shared/models/http/http-error-codes.ts:7]: Please keep the `const` here that reduce client bundle size

I'm not sure the HTTP status label is very interesting. If you think so, you'll have to build the object yourself

Comment (fflorent) [reply shared/models/http/http-error-codes.ts:7]: Yeah, that was the price for this feature. If you don't think it is worth it to mention the label, I revert that.

Comment (fflorent) [reply shared/models/http/http-error-codes.ts]: Done

**Comment by Chocobozzz** [comment]
Thanks!

**Pull request merged**